internal/codegen/tpl_file.go

```diff
@@ -105,6 +110,24 @@ func (f *TplFile) Static() (out string, err error) {
 	return "", nil
 }
 
+// Once will only generate this file a single time, and store that fact
+//
+// The first time a Once file is generated, it has its provenance stored
+// in the stencil.lock file.  Going forward, Once checks the lock file
+// for history of the file, and if it finds it, it performs the same
+// action as file.Skip.
+//
+//	{{ file.Once }}
+func (f *TplFile) Once() (out string, err error) {
```

Comment:
Just thinking out loud here - should you add an optional key argument to this or something? That way the author of the file can invalidate it in addition to the client being able to?

Reply:
I don't think it's particularly valid to version a Once file.  Like, if you only ever want it to be one thing, it's gotta be a suggested first whack at something.  Overwriting later if the key changes seems like they shouldn't be using once, they should be using blocks.  Did you have a scenario in mind here?

Reply:
If you just remove file.once and put file.delete, it'll switch to that behavior.  file.once is still executed and evaluated on every run, it just turns into skip if it passes the logic.

Maybe that's a thing, but it's hard to say.  It feels like that's going to be a more-normal template if you're gonna rev it.  If people want a new model for it they'd delete and re-generate.  I'd almost say just use file.Static for that, if you know the file needs to exist in soem fashion and you want to seed it with something.  I think we'll start with this, and see if we need to add some more extension semantics to it later (optional version key).